tests/hip/test_hip_copy.cpp

```diff
@@ -19,22 +19,74 @@
 #include <gtest/gtest.h>
 
 // System include(s).
+#include <memory>
 #include <tuple>
 
 // Objects used in the test(s).
 static vecmem::hip::host_memory_resource hip_host_resource;
 static vecmem::memory_resource* hip_host_resource_ptr = &hip_host_resource;
 static vecmem::hip::device_memory_resource hip_device_resource;
-static vecmem::memory_resource* hip_device_resource_ptr = &hip_device_resource;
+static vecmem::memory_resource* hip_device_resource_ptr =
+    &hip_device_resource;
+static vecmem::hip::managed_memory_resource hip_managed_resource;
+static vecmem::memory_resource* hip_managed_resource_ptr =
+    &hip_managed_resource;
 static vecmem::copy hip_host_copy;
 static vecmem::copy* hip_host_copy_ptr = &hip_host_copy;
 static vecmem::hip::copy hip_device_copy;
 static vecmem::copy* hip_device_copy_ptr = &hip_device_copy;
+static std::unique_ptr<vecmem::hip::stream_wrapper> hip_stream;
+static std::unique_ptr<vecmem::hip::async_copy> hip_async_device_copy;
+static vecmem::copy* hip_async_device_copy_ptr = nullptr;
 
-/// The configurations to run the tests with.
-static const auto hip_copy_configs =
-    testing::Values(std::tie(hip_device_copy_ptr, hip_host_copy_ptr,
-                             hip_device_resource_ptr, hip_host_resource_ptr));
+/// Environment taking care of setting up and tearing down the async test types
+class AsyncHIPCopyEnvironment : public ::testing::Environment {
+public:
+    /// Set up the environment
+    void SetUp() override {
+        hip_stream = std::make_unique<vecmem::hip::stream_wrapper>();
+        hip_async_device_copy =
+            std::make_unique<vecmem::hip::async_copy>(*hip_stream);
+        hip_async_device_copy_ptr = hip_async_device_copy.get();
+    }
+    /// Tear down the environment
+    void TearDown() override {
+        hip_async_device_copy.reset();
+        hip_stream.reset();
+        hip_async_device_copy_ptr = nullptr;
+    }
+};
+
+/// Register the environment
+static ::testing::Environment* const async_hip_copy_env =
+    ::testing::AddGlobalTestEnvironment(new AsyncHIPCopyEnvironment{});
+
+/// The configurations to run the tests with. Skip tests with asynchronous
```

Comment:
Not sure if Windows fails here? I have no way to test it but we may as well define the tests for Windows in case someone does...

Reply:
While HIP is now technically available on Windows, we don't test any of the GPU libraries on Windows in the CI. As it's really not that easy to install CUDA, oneAPI, or any of these things as part of the CI. (Not impossible as far as I can tell, but far from easy.)

So we don't need to worry about any of that in the CUDA, HIP and SYCL libraries.